WalletWasabi/Tor/Http/IHttpClient.cs

```diff
@@ -1,16 +1,43 @@
 using System;
 using System.Net.Http;
-using System.Security.Authentication;
 using System.Threading;
 using System.Threading.Tasks;
 
 namespace WalletWasabi.Tor.Http
 {
 	/// <summary>
-	/// Interface defining HTTP client capable of sending HTTP requests.
+	/// Interface defining HTTP client capable of sending HTTP requests that are relative to some base URI.
+	/// <para>This is useful, for example, to send requests to Wasabi Backend server.</para>
 	/// </summary>
 	public interface IHttpClient
 	{
+		Func<Uri> DestinationUriAction { get; }
```

Comment:
This is one thing that requires a careful design decision: Either you allow the function to return `null` or not. Both can be fine but it affects how people can use the API.

* If we decide to allow `Uri?`, then an implementation of `IHttpClient` can do absolute HTTP requests.
* If we decide to allow only `Uri`, then an API consumer must provide some base URI and and such, it's slightly more restrictive in its consumption.

Reply:
It is the same as it was defined in IRelativeHttpClient before.

Reply:
MS created a [HttpClientHandler](https://github.com/microsoft/referencesource/blob/5697c29004a34d80acdaf5742d7e699022c64ecd/System/net/System/Net/Http/HttpClientHandler.cs#L25)

which is a [HttpMessageHandler](https://github.com/microsoft/referencesource/blob/5697c29004a34d80acdaf5742d7e699022c64ecd/System/net/System/Net/Http/HttpMessageHandler.cs#L9) that has a method of 

        protected internal abstract Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken);

This is kind of similar that your IHttpClient.SendAsync.

Reply:
`IHttpClient.SendAsync.` is indeed inspired by [HttpClient](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.sendasync?view=net-5.0) API.

> It is the same as it was defined in IRelativeHttpClient before.

It was in `IRelativeHttpClient` as `Func<Uri> DestinationUriAction { get; }` because all `IRelativeHttpClient` are supposed to have some base URI which is used in all relative HTTP requests. When you put it to `IHttpClient`, it should be clear how people are supposed to use it. Is it mandatory to specify base URI or not? This affects many other design decisions.